File: jphp-ssh-client-ext/src/main/resources/JPHP-INF/sdk/php/ssh/client/Connection.php
<?php

namespace php\ssh\client;

/**
 * Class Connection
 * @package php\ssh\client
 */
abstract class Connection
{
	/**
     * description
     * @return Commander
     */
	public function execute() {
	}
	
	/**
     * Close active connection
	 * @return void
     */
	public function close() {
    }
	
	/**
     * description
	 * @return string
     */
    public function getHost() {
    }
	
	/**
     * description
	 * @return int
     */
    public function getPort() {
    }
}

File: 3rd-party/ssh-client/src/main/java/org/artemir/ssh/client/Connection.java
package org.artemir.ssh.client;

import com.jcraft.jsch.*;

public class Connection {
    private String host;
    private int port;
    private Session session;
    private Auth auth;
    private Commander commander;

    public Connection(String address, int port, Auth auth) {
        this.host = address;
        this.port = port;
        this.auth = auth;
    }

    public Commander execute() throws JSchException {
        if (session == null) {
            session = auth.createSession(host, port);
            session.setConfig("StrictHostKeyChecking", "no");
            commander = new Commander(this);
        }

        if (!session.isConnected()) {
            session.connect();
        }

        return commander;
    }

    public Session getSession() {
        return session;
    }

    public String getHost() {
        return host;
    }

    public int getPort() {
        return port;
    }

    public Auth getAuth() {
        return auth;
    }

    public Commander getCommander() {
        return commander;
    }

    public void close() {
        if (session != null && session.isConnected()) {
            session.disconnect();
        }
    }
}
